src/main/java/org/testng/TestRunner.java

```diff
@@ -514,16 +511,9 @@ private void initMethods() {
             m_excludedMethods,
             comparator);
 
-    m_allTestMethods =
-        MethodHelper.collectAndOrderMethods(
-            testMethods,
-            true /* forTest? */,
-            m_runInfo,
-            m_annotationFinder,
-            false /* unique */,
-            m_excludedMethods,
-            comparator);
-    m_classMethodMap = new ClassMethodMap(Arrays.asList(m_allTestMethods), m_xmlMethodSelector);
+    testMethodsContainer = new TestMethodContainer(getAllTestMethods());
```

Comment:
Tips: Could be lazy-loaded with a Function: `() -> getAllTestMethods()`

Reply:
I am not sure if we are going to be getting a lot of help with lazy loading.

Reply:
You are right. It was just for fun ;)